test/blackbox/common/DDSBlackboxTestsContentFilter.cpp

```diff
@@ -495,6 +502,55 @@ TEST_P(DDSContentFilter, WithLimitsSeveralReaders)
     test_run(reader, state, 2u);
 }
 
+TEST_P(DDSContentFilter, WithLimitsDynamicReaders)
+{
+    TestState state;
+
+    // Only one filtered reader created
+    auto reader = prepare_test(state, fastrtps::ResourceLimitedContainerConfig::fixed_size_configuration(2u), 0u);
+    ASSERT_NE(nullptr, reader);
+
+    // We want a single filter to be applied, and check only for reader discovery changes
+    state.set_filter_expression("index BETWEEN %0 AND %1", { "2", "4" });
+
+    std::cout << "========= First reader =========" << std::endl;
+    state.send_data(reader, filter_counter, 3u, { 2, 3, 4 }, true, 1u);
+
+    // Adding a second reader should increase the number of writer filters
+    std::cout << "========= Create a second reader =========" << std::endl;
+    auto reader_2 = state.create_filtered_reader();
+    ASSERT_NE(nullptr, reader_2);
+
+    state.writer.wait_discovery(3);
+    std::this_thread::sleep_for(std::chrono::milliseconds(250));
```

Comment:
Maybe a note here stating that this wait is to prevent the discovery data race would be helpful